Docker development  image build fails

Running `gulp developImageBuild` fails with:

```

#11 67.20
#11 67.20 > amp-dev@1.0.5 postinstall
#11 67.20 > gulp downloadTestResources && husky install
#11 67.20

#11 68.29 [7:28:58 PM] › ℹ  info      Config: environment=development test=false

#11 69.58 fatal: not a git repository (or any of the parent directories): .git

#11 69.59 Error: Command failed: git log -1 --pretty=format:%h
#11 69.59 fatal: not a git repository (or any of the parent directories): .git
#11 69.59
#11 69.59     at checkExecSyncError (child_process.js:643:11)
#11 69.59     at execSync (child_process.js:679:15)
#11 69.59     at Object.module.exports.version (/amp-dev/platform/lib/utils/git.js:19:10)
#11 69.59     at Object.<anonymous> (/amp-dev/gulpfile.js/deploy.js:39:54)
#11 69.59     at Module._compile (internal/modules/cjs/loader.js:1085:14)

#11 69.59     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
#11 69.59     at Module.load (internal/modules/cjs/loader.js:950:32)
#11 69.59     at Function.Module._load (internal/modules/cjs/loader.js:790:14)
#11 69.59     at Module.require (internal/modules/cjs/loader.js:974:19)
#11 69.59     at require (internal/modules/cjs/helpers.js:92:18)
#11 69.59     at Array.map (<anonymous>)
#11 69.59     at Object.<anonymous> (/amp-dev/gulpfile.js/index.js:28:4)
#11 69.59     at Module._compile (internal/modules/cjs/loader.js:1085:14)
#11 69.59     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
#11 69.59     at Module.load (internal/modules/cjs/loader.js:950:32)
#11 69.59     at Function.Module._load (internal/modules/cjs/loader.js:790:14) {
#11 69.59   status: 128,
#11 69.59   signal: null,
#11 69.59   output: [
#11 69.59     null,
#11 69.59     <Buffer >,
#11 69.59     <Buffer 66 61 74 61 6c 3a 20 6e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
#11 69.59   ],
#11 69.59   pid: 1366,
#11 69.59   stdout: <Buffer >,
#11 69.59   stderr: <Buffer 66 61 74 61 6c 3a 20 6e 6f 74 20 61 20 67 69 74 20 72 65 70 6f 73 69 74 6f 72 79 20 28 6f 72 20 61 6e 79 20 6f 66 20 74 68 65 20 70 61 72 65 6e 74 20 ... 19 more bytes>
#11 69.59 }
#11 69.61 npm ERR! code 1
#11 69.61 npm ERR! path /amp-dev
#11 69.61 npm ERR! command failed
#11 69.61 npm ERR! command sh -c gulp downloadTestResources && husky install
#11 69.65
#11 69.65 npm ERR! A complete log of this run can be found in:
#11 69.65 npm ERR!     /root/.npm/_logs/2021-07-23T19_28_59_842Z-debug.log

#11 ERROR: executor failed running [/bin/sh -c npm install]: exit code: 1
------
 > [7/8] RUN npm install:
------

executor failed running [/bin/sh -c npm install]: exit code: 1

[21:28:59] 'developImageBuild' errored after 1.28 min
[21:28:59] Error: docker process exited with code 1
    at ChildProcess.<anonymous> (/Users/sbenz/Projects/amp.dev/platform/lib/utils/sh.js:66:16)
    at ChildProcess.emit (events.js:375:28)
    at ChildProcess.emit (domain.js:532:15)
    at maybeClose (internal/child_process.js:1055:16)
    at Socket.<anonymous> (internal/child_process.js:441:11)
    at Socket.emit (events.js:375:28)
    at Socket.emit (domain.js:532:15)
    at Pipe.<anonymous> (net.js:675:12)
    at Pipe.callbackTrampoline (internal/async_hooks.js:131:17)
```

--- a/platform/lib/utils/git.js
+++ b/platform/lib/utils/git.js
@@ -19,6 +19,8 @@ module.exports.version = () => {
   try {
     return execSync('git log -1 --pretty=format:%h ').toString().trim();
   } catch (e) {
+    // This method is called from gulpfile.js/deploy.js even if in a
+    // non-git context, like the development Docker image
     return 'detached';
   }
 };

--- a/platform/lib/utils/git.js
+++ b/platform/lib/utils/git.js
@@ -16,7 +16,11 @@
 const {execSync} = require('child_process');
 
 module.exports.version = () => {
-  return execSync('git log -1 --pretty=format:%h ').toString().trim();
+  try {
+    return execSync('git log -1 --pretty=format:%h ').toString().trim();
+  } catch (e) {
+    return 'detached';
+  }
 };
 
 module.exports.message = () => {

